### PR TITLE
feat(resolver): move feature-flag.resolver from fabric8-ui to ngx-feature-flag

### PR DESCRIPTION
**Pull request by rohitkrai03**
Fixes https://github.com/openshiftio/openshift.io/issues/4116
Fixes https://github.com/fabric8-services/fabric8-auth/issues/570

- `ngx-feature-flag` library was extracted out of `fabric8-ui` so that external libraries such as `fabric8-planner` would be able to use feature flags.
- But there is still one piece of code `FeatureFlagResolver` inside `fabric8-ui` which needs to move to `ngx-feature-flag` so that the other libraries do not need to write it all over again.

### CODE REVIEW

**Comment by rohitkrai03** [comment]
@andrewazores Please review.

File: src/app/resolver/feature-flag.resolver.ts
Comment (andrewazores) [inline src/app/resolver/feature-flag.resolver.ts]: It looks like something weird was happening with the indentation in this file, even before it was moved across repos. Could you reindent it please since we're already here?

File: src/app/feature-flag.module.ts
Comment (andrewazores) [inline src/app/feature-flag.module.ts]: Hmm. Should this be in `providers`, or just `exports`? I think I'd rather have this module simply export the resolver, and consumers decide where and when to inject it into their DI hierarchy. FWIW the fabric8-ui app.module already has the resolver in its own `providers` array at the top level of the application.

**Comment by rohitkrai03** [comment]
@andrewazores Done.

**Review by andrewazores — APPROVED** [review]

**Comment by rohitkrai03** [comment]
@andrewazores Do we need to do npm publish with new version for this?

**Comment by andrewazores** [comment]
@rohitkrai03 I think so, yes

**Comment by rohitkrai03** [comment]
So, who does that for this repo? Also, it says only reviews by reviewers with write access counts.

**Comment by andrewazores** [comment]
Yea, I don't have write access on this repo. I think it's normally just Corinne who does the releases here.

**Comment by rohitkrai03** [comment]
Is there anyone else who can do this if Corinne is not available? @joshuawilson?

**Review by joshuawilson — APPROVED** [review]
looks like this copy only has formatting changes, LGTM

**Pull request merged**
